nifi-registry/nifi-registry-core/nifi-registry-web-api/src/main/java/org/apache/nifi/registry/web/security/authentication/csrf/CsrfRequestMatcher.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.registry.web.security.authentication.csrf;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.security.web.csrf.CsrfFilter;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+
+import static org.springframework.web.util.WebUtils.getCookie;
+
+/**
+ * Cross-Site Request Forgery Mitigation Request Matcher
+ */
+public class CsrfRequestMatcher implements RequestMatcher {
+
+    @Override
+    public boolean matches(final HttpServletRequest request) {
+        final boolean matches;
+
+        if (CsrfFilter.DEFAULT_CSRF_MATCHER.matches(request)) {
+            // Presence of Request Token Cookie requires invoking the CsrfFilter
+            final Cookie requestTokenCookie = getCookie(request, CsrfCookieName.REQUEST_TOKEN.getCookieName());
+            matches = requestTokenCookie != null;
+        } else {
+            matches = false;
+        }
+
+        return matches;
+    }
```

Comment:
By default when declaring `boolean matches` it is already false. Setting to false I would think only clarifies its actual value. Hence the code could be simplified with

```suggestion
        final boolean matches = false;

        if (CsrfFilter.DEFAULT_CSRF_MATCHER.matches(request)) {
            // Presence of Request Token Cookie requires invoking the CsrfFilter
            final Cookie requestTokenCookie = getCookie(request, CsrfCookieName.REQUEST_TOKEN.getCookieName());
            matches = requestTokenCookie != null;
        } 

        return matches;
    }
```

Reply:
> By default when declaring `boolean matches` it is already false. Setting to false I would think only clarifies its actual value. Hence the code could be simplified with

Thanks for the feedback. The `matches` variable is declared `final`, so that is the reason for the explicit `false` assignment, which would otherwise be disallowed unless the `final` keyword were removed.

Reply:
Thanks for clarifying that. Just curios, what is the advantage of using `final` here?

Reply:
There is not a great benefit in this particular case with a small method and minimal conditionals, but it is more of a general approach to use `final` as the standard, and only define re-assignable variables when absolutely necessary.